File: nan21.dnet.module.ad/nan21.dnet.module.ad.ui.extjs/src/main/webapp/src/ds/AttachmentTypeLovDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.ad.data.ds.model.AttachmentTypeLovDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"clientId", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"description", type:"string"},
		{name:"active", type:"boolean"},
		{name:"useInContext", type:"string"}
	]
});

Ext.define("net.nan21.dnet.module.ad.data.ds.model.AttachmentTypeLovDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"description", type:"string"},
		{name:"active", type:"boolean", useNull:true},
		{name:"useInContext", type:"string"}
	] 
});
	
Ext.define("net.nan21.dnet.module.ad.data.ds.param.AttachmentTypeLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 

File: nan21.dnet.module.ad/nan21.dnet.module.ad.presenter/src/main/java/net/nan21/dnet/module/ad/data/ds/model/AttachmentTypeLovDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.ad.data.ds.model;

import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeLov;

import net.nan21.dnet.module.ad.data.domain.entity.AttachmentType;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = AttachmentType.class, jpqlWhere = " e.active = true ", sort = { @SortField(field = AttachmentTypeLovDs.fNAME) })
public class AttachmentTypeLovDs extends AbstractTypeLov<AttachmentType> {

    public static final String fUSEINCONTEXT = "useInContext";

    @DsField(jpqlFilter = " :useInContext like e.useInContext ")
    private String useInContext;

    public AttachmentTypeLovDs() {
        super();
    }

    public AttachmentTypeLovDs(AttachmentType e) {
        super(e);
    }

    public String getUseInContext() {
        return this.useInContext;
    }

    public void setUseInContext(String useInContext) {
        this.useInContext = useInContext;
    }

}
